cmd/destroy.go

```diff
@@ -156,26 +161,37 @@ func deleteEntriesFromHostsFile(containers []types.GenericContainer, bridgeName
 func destroyLab(ctx context.Context, c *clab.CLab) (err error) {
 
 	labels := []*types.GenericFilter{{FilterType: "label", Match: c.Config.Name, Field: "containerlab", Operator: "="}}
-	containers, err := c.Runtime.ListContainers(ctx, labels)
+	containers, err := c.ListContainers(ctx, labels)
 	if err != nil {
-		return fmt.Errorf("could not list containers: %v", err)
-	}
-	if len(containers) == 0 {
-		return nil
+		return err
 	}
 
-	// get lab directory used by this lab to remove it later if cleanup is used
 	var labDir string
+	for _, ctr := range containers {
+		if n, ok := c.Nodes[ctr.ShortID]; ok {
+			c.Nodes[ctr.ShortID] = n
+			labDir = n.Config().LabDir
+		}
+	}
+
```

Comment:
What was the idea here to swap the `containers[0].Labels["clab-node-lab-dir"]` as a way to derive the lab directory?

Reply:
hm... i don't remember doing this. I'll revert back to how it is now in main.